scripts/autonomous_cross_deduplicator_par.py

```diff
@@ -0,0 +1,151 @@
+    parser.add_argument('--from-dir', '-f',
+                        help='the earlies batch to work on. Only use this if'
+                             'other tasks are working on the preceding'
+                             'batches, otherwise it will stall forever.')
+    parser.add_argument('--upto-dir', '-u',
+                        help='the latest batch to work on.')
+    parser.add_argument('--temp-dir', '-T', type=Path,
+                        help='the directory used to temporarily store partial '
+                             'results')
+    parser.add_argument('--permutations', '-p', type=int, default=256,
+                        help='the number of permutations per paragraph (256).')
+    parser.add_argument('--threshold', '-t', type=float, default=0.9,
+                        help='the Jaccard similarity threshold (0.9).')
+    parser.add_argument('--processes', '-P', type=int, default=1,
+                        help='number of worker processes to use (max is the '
+                             'num of cores, default: 1). Note that in order '
+                             'to deduplicate documents, much memory might be '
+                             'needed, so it is a good idea to be conservative '
+                             'with the number of processes.')
+    parser.add_argument('--log-level', '-L', type=str, default='info',
+                        choices=['debug', 'info', 'warning',
+                                 'error', 'critical'],
+                        help='the logging level.')
+    args = parser.parse_args()
+    if not args.output_dir.is_dir():
+        parser.error('The directory for the batches must exist.')
+    if args.temp_dir:
+        if not args.temp_dir.is_dir():
+            parser.error('The temporary directory, if set, must exist.')
+    return args
+
+
+def has_minhash_content(directory: Path) -> bool:
+    """
+    Tells whether a given directory contains all the following files or not:
+    1.doc_ids, 1.files, 1.minhashes
+    """
+    return ((directory / '1.doc_ids').is_file()
+            and (directory / '1.files').is_file()
+            and (directory / '1.minhashes').is_file())
+
+
+def assemble_targets(input_dir: Path, output_dir: Path,
+                     from_dir: Path, upto_dir):
+    """
+    Returns a list of tuples. The tuples contain the parameters for the
+    deduplicate_other() method:
+    The input (minhash self) dir of the target
+    The list of dirs of older batches (minhash full)
+    The output (minhash full) dir for the target.
+
+    Ignores dirs which do not contain the required files
+    """
+    full_list = [dir.name for dir in sorted(input_dir.iterdir())
+                 if has_minhash_content(dir)]
+    if upto_dir:
+        try:
+            upto_i = full_list.index(upto_dir)
+        except ValueError:
+            logging.error('The upto-dir is not a valid batch')
+            raise
+        full_list = full_list[: upto_i+1]

```

Comment:
I think flake8 would prefer `[:upto_i + 1]` instead.

Reply:
Also, I wouldn't overwrite the `full_list` variable, because the name isn't true after that. You can also index into a slice, so...

Reply:
flake8 had no problem with that syntax, but I changed it.

I renamed the variable.
Overwriting it is important, because that part of the code is conditional.